Scale job badges to fit custom titles of any length

analytically opened the pull request:
Badge endpoint now dynamically calculates width for custom titles with length-based padding (12px/6px/0px). Preserves original status section widths and maintains backward compatibility with default badges.

Fixes #9237

analytically commented:
<img width="382" height="122" alt="screentshot-2025-10-06 at 11 22 25@2x" src="https://github.com/user-attachments/assets/c6f1e0d4-36cb-41dd-a824-ae5690f15477" />
<img width="770" height="144" alt="screentshot-2025-10-06 at 11 22 42@2x" src="https://github.com/user-attachments/assets/61b0ce86-2f56-4ee6-aba9-50bbcd21bb98" />

taylorsilva commented on atc/api/jobserver/badge_test.go:
```diff
@@ -0,0 +1,251 @@
+package jobserver_test
```
Existing tests for the badge endpoint are here: https://github.com/concourse/concourse/blob/97393cea938d4dfb125071b9542b9e151eab4a47/atc/api/jobs_test.go#L533

I think most of your tests here are already covered by the existing tests. Can you move your tests alongside the existing tests for the badge endpoint?

taylorsilva reviewed (changes requested):
Existing badge endpoint tests are failing:

```
$ ginkgo -r -p ./atc/api
...
Summarizing 2 Failures:
  [FAIL] Jobs API GET /api/v1/teams/:team_name/pipelines/:pipeline_name/jobs/:job_name/badge when authorized when generates bagde title [It] html escapes title
  /Users/taylor/workspace/concourse/concourse/atc/api/jobs_test.go:644
  [FAIL] Jobs API GET /api/v1/teams/:team_name/pipelines/:pipeline_name/jobs/:job_name/badge when authorized when generates bagde title [It] uses url `title` parameter
  /Users/taylor/workspace/concourse/concourse/atc/api/jobs_test.go:608

Ran 1003 of 1003 Specs in 3.472 seconds
FAIL! -- 1001 Passed | 2 Failed | 0 Pending | 0 Skipped


Ginkgo ran 9 suites in 9.932889291s

There were failures detected in the following suites:
  api ./atc/api
```

analytically committed:
Fix badge scaling for custom titles

Badge endpoint now dynamically calculates width for custom titles with
length-based padding (12px/6px/0px). Preserves original status section
widths and maintains backward compatibility with default badges.

Fixes #9237

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>

taylorsilva reviewed (approved):
LGTM! 

<img width="1086" height="64" alt="image" src="https://github.com/user-attachments/assets/3d1cc652-fe76-4bcd-9e22-b7f91e1f3e30" />
<img width="1054" height="64" alt="image" src="https://github.com/user-attachments/assets/47d8e89a-0717-4a66-9dc1-9255511df92a" />
<img width="1029" height="64" alt="image" src="https://github.com/user-attachments/assets/ac73b36f-0cff-4604-a108-fa92524fea95" />
<img width="1054" height="62" alt="image" src="https://github.com/user-attachments/assets/f19d1e02-05c0-4da9-8368-25ace00aba53" />
<img width="1060" height="61" alt="image" src="https://github.com/user-attachments/assets/89af4df3-48b3-49ef-a594-044676d5c964" />
